pkg/utils/node/predicates.go

```diff
@@ -42,13 +49,37 @@ func IsPastTTL(node *v1.Node) bool {
 	return time.Now().After(ttlTime)
 }
 
-// TODO: implement underutilized function (some generalized predicate)
-func IsUnderutilized(pods []*v1.Pod) bool {
-	counter := 0
+// IsUnderutilized returns true if all of a node's pod's resource requests sum to less than the given thresholds
+func IsUnderutilized(node *v1.Node, pods []*v1.Pod) bool {
+	threshold := v1.ResourceList{
+		v1.ResourceCPU: *resource.NewMilliQuantity(
+			roundQuantity(node.Status.Capacity.Cpu().MilliValue(), UnderutilizedPercentage),
```

Comment:
I'm not sure if we have to tackle it right now, but I do think people are likely to want to set different thresholds for the different resources. Eg they might want 40% CPU utilization to be considered "underutilization", but 40% memory utilization to be fine (say they're concerned about using more memory over time or worried that they occasionally need spikes of memory)?

Reply:
I can see people wanting that use case as well. Like Ellis said, I'll hold back off on doing the underutilization until we have a discussion about it.